examples/ticks/date_index_formatter.py

```diff
@@ -0,0 +1,80 @@
+"""
+=====================================
+Custom tick formatter for time series
+=====================================
+
+.. redirect-from:: /gallery/text_labels_and_annotations/date_index_formatter
+.. redirect-from:: /gallery/ticks/date_index_formatter2
+
+When plotting daily data, e.g., financial time series, one often wants
+to leave out days on which there is no data, for instance weekends, so that
+the data are plotted at regular intervals without extra spaces for the days
+with no data.
+The example shows how to use an 'index formatter' to achieve the desired plot.
+"""
+
+import numpy as np
+import matplotlib.pyplot as plt
+import matplotlib.cbook as cbook
+import matplotlib.lines as ml
+from matplotlib.dates import DateFormatter, DayLocator
+from matplotlib.ticker import Formatter
+
+
+# Load a numpy record array from yahoo csv data with fields date, open, high,
+# low, close, volume, adj_close from the mpl-data/sample_data directory. The
+# record array stores the date as an np.datetime64 with a day unit ('D') in
+# the date column.
+r = (cbook.get_sample_data('goog.npz', np_load=True)['price_data']
+     .view(np.recarray))
+r = r[:9]  # get the first 9 days
+
+fig, (ax1, ax2) = plt.subplots(nrows=2, figsize=(6, 6),
+                               constrained_layout={'hspace': .15})
+
+# First we'll do it the default way, with gaps on weekends
+ax1.plot(r.date, r.adj_close, 'o-')
+
+# Highlight gaps in daily data
+gaps = np.flatnonzero(np.diff(r.date) > np.timedelta64(1, 'D'))
+for gap in r[['date', 'adj_close']][np.stack((gaps, gaps + 1)).T]:
+    ax1.plot(gap.date, gap.adj_close, 'w--', lw=2)
+ax1.legend(handles=[ml.Line2D([], [], ls='--', label='Gaps in daily data')])
+
+ax1.set_title("Plot y at x Coordinates")
+ax1.xaxis.set_major_locator(DayLocator())
+ax1.xaxis.set_major_formatter(DateFormatter('%a'))
+
+
+# Next we'll write a custom index formatter
+def format_date(x, _):
```

Comment:
```suggestion
# Next we'll write a custom index formatter. Below we will plot 
# the data against an index that goes from 0, 1,  ... len(data).  Instead of 
# formatting the tick marks as integers, we format as times.
def format_date(x, _):
```

Reply:
fixed